packages/core/src/dispatcher.ts

```diff
@@ -45,6 +47,8 @@ export default class Dispatcher extends EventEmitter {
 
   private logger: Logger
   private _isRunning = true
+  private _messageHandler = this.handleMessage.bind(this)
```

Comment:
What is `this` here? Seems like it's outside the scope of the class instance.

Reply:
just check and it does seem this is properly scoped to the instance. i.e. state set in the constructor is accessible when `handleMessage` gets called.

but I'd probably prefer something even simpler which would be to remove this in favor of an arrow function which achieves the same thing

```js
handleMessage = async (message: any): Promise<void> => {
```